api/client.go

```diff
@@ -282,6 +282,10 @@ func (c *Client) AbortCurrentUpgrade() error {
 
 // FindTools returns a List containing all tools matching the specified parameters.
 func (c *Client) FindTools(majorVersion, minorVersion int, series, arch, agentStream string) (result params.FindToolsResult, err error) {
+	if c.facade.BestAPIVersion() == 1 && agentStream != "" {
+		return params.FindToolsResult{}, errors.New(
+			"passing agent-stream not supported by target model")
```

Comment:
It's really the controller that doesn't support it, since the API server is shared across the models. (This might be a bit pedantic.)